### PR TITLE
Added support for identifying a car by Player.

### PR DESCRIPTION
**Pull request by BeeHiveJava**

### CODE REVIEW

File: Soccar [Physics]/src/nl/soccar/physics/models/BallPhysics.java [hunk 1]
```diff
@@ -65,9 +65,15 @@ public void step() {
 
     @Override
     public void reset() {
-        body.setLinearVelocity(new Vec2(0.0F, 0.0F));
-        body.setAngularVelocity(0.0F);
-        body.setTransform(originalPos, 0.0F);
+        setPosition(originalPos.x, originalPos.y, 0, body.getLinearVelocity().x, body.getLinearVelocity().y, body.getAngularVelocity());
+    }
+
+    public void setPosition(float x, float y, float degree, float linearVelocityX, float linearVelocityY, float angularVelocity) {
```
Comment (ghost) [inline Soccar [Physics]/src/nl/soccar/physics/models/BallPhysics.java:71]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java [hunk 2]
```diff
@@ -104,11 +103,17 @@ public void step() {
...
+
+    public void setPosition(float x, float y, float degree, float linearVelocityX, float linearVelocityY, float angularVelocity) {
+        car.move(x, y, degree);
+
+        body.setLinearVelocity(new Vec2(linearVelocityX, linearVelocityY));
+        body.setAngularVelocity(angularVelocity);
+        body.setTransform(new Vec2(x, y), (float) Math.toRadians(degree));
 
-        wheels.forEach(WheelPhysics::reset);
+        //wheels.forEach(WheelPhysics::reset);
```
Comment (ghost) [inline Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java:116]: MAJOR This block of commented-out lines of code should be removed. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java [hunk 3]
```diff
@@ -104,11 +103,17 @@ public void step() {
 
     @Override
     public void reset() {
-        body.setLinearVelocity(new Vec2(0.0F, 0.0F));
-        body.setAngularVelocity(0.0F);
-        body.setTransform(originalPos, originalDegree);
+        setPosition(originalPos.x, originalPos.y, originalDegree, 0, 0, 0);
+    }
+
+    public void setPosition(float x, float y, float degree, float linearVelocityX, float linearVelocityY, float angularVelocity) {
```
Comment (ghost) [inline Soccar [Physics]/src/nl/soccar/physics/models/CarPhysics.java:109]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Physics]/src/nl/soccar/physics/GameEngine.java [hunk 4]
```diff
@@ -127,18 +130,58 @@ private void checkScored() {
...
+        if (object instanceof BallPhysics) {
+            ball = null;
+        }
+
         synchronized (objects) {
             objects.remove(object);
         }
     }
 
+    public CarPhysics getCarFromPlayer(Player player) {
```
Comment (ghost) [inline Soccar [Physics]/src/nl/soccar/physics/GameEngine.java:177]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Physics]/src/nl/soccar/physics/GameEngine.java [hunk 5]
```diff
@@ -127,18 +130,58 @@ private void checkScored() {
         }
     }
 
+    public void addCar(Player player, CarPhysics car) {
```
Comment (ghost) [inline Soccar [Physics]/src/nl/soccar/physics/GameEngine.java:133]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Physics]/src/nl/soccar/physics/GameEngine.java [hunk 6]
```diff
@@ -127,18 +130,58 @@ private void checkScored() {
...
 
+    public void addCar(Player player, CarPhysics car) {
+        cars.put(player, car);
+
+        synchronized (objects) {
+            objects.add(car);
+        }
+    }
+
+    public void removeCar(Player player) {
```
Comment (ghost) [inline Soccar [Physics]/src/nl/soccar/physics/GameEngine.java:141]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by ghost** [comment]
SonarQube analysis reported 7 issues
* MAJOR 2 major
* MINOR 5 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [GameEngine.java#L110](https://github.com/PTS3-S34A/Physics/blob/08f28f15e3dae1027cb02a6de36d02b8e00f06db/Soccar [Physics]/src/nl/soccar/physics/GameEngine.java#L110): Rename "ball" which hides the field declared at line 32. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AHiddenFieldCheck)

**Review by Ranray — APPROVED** [review]

**Pull request merged**